integration-test/playwright/page-objects/duckplayer-overlays.js

```diff
@@ -33,6 +33,22 @@ const userValues = {
     }
 }
 
+// Possible UI Settings
+const uiSettings = {
+    /** @type {import("../../../src/features/duck-player.js").UISettings} */
+    'default overlay copy': {
+        overlayCopy: 'default'
+    },
+    /** @type {import("../../../src/features/duck-player.js").UISettings} */
+    'overlay copy a1': {
+        overlayCopy: 'a1'
+    },
+    /** @type {import("../../../src/features/duck-player.js").UISettings} */
+    'overlay copy b1': {
+        overlayCopy: 'b1'
+    }
+}
+
```

Comment:
You can annotate the object to be a `Record<string, import("../../../src/features/duck-player.js").UISettings>` instead to make this simpler

Reply:
Yeah I tried it at first but then autocomplete didn’t work as well for some reason. Will try it again and see.

Reply:
Maybe I’m doing something wrong, but when I use `Record<string…` then set `@param {keyof uiSettings} uiSetting` in a function further down, it doesn’t pick up on the keys and defers to the type string in the Record instead. Screenshots below

Using individual type declarations 
<img width="868" alt="Screenshot 2024-07-10 at 15 09 00" src="https://github.com/duckduckgo/content-scope-scripts/assets/115609/d42a4c14-bd91-4e01-8e9c-37d0a93b10ae">

Using the top level Record<> declaration
<img width="879" alt="Screenshot 2024-07-10 at 15 10 13" src="https://github.com/duckduckgo/content-scope-scripts/assets/115609/5412c3f1-b418-4c26-a1c0-f6f931ad38c5">

The main reason I went with individual type declarations is that I noticed you used the same pattern just a few lines above, in `userValues` and I figured it was for the same reason (to get around this keyof limitation)

Reply:
The following screenshots summarise my problem a bit better

<img width="1068" alt="image" src="https://github.com/duckduckgo/content-scope-scripts/assets/115609/84e10687-9790-40b3-9aac-b469c5613a4c">

<img width="1127" alt="image" src="https://github.com/duckduckgo/content-scope-scripts/assets/115609/55f9e765-76ca-41d9-877c-85a897670a3a">